apps/enterprise-knowledge-retrieval/enterprise_knowledge_retrieval.ipynb

```diff
@@ -0,0 +1,1750 @@
+       " ('- What role did Marie Curie play in the discovery of radioactivity?',\n",
+       "  'Unable to answer question'),\n",
+       " ('- How did the Space Race impact advancements in technology?',\n",
+       "  'Agent stopped due to iteration limit or time limit.'),\n",
+       " ('- What are the primary differences between the Baroque and Classical music periods?',\n",
+       "  'The Baroque period was characterized by a more ornate and complex style, while the Classical period was known for its simplicity and symmetry. Classical musicians focused on musical analysis and harmony and counterpoint. The greatest composers of the Baroque period include Claudio Monteverdi, Antonio Vivaldi, Johann Sebastian Bach, and George Frideric Handel, while the greatest composers of the Classical period include Joseph Haydn, Wolfgang Amadeus Mozart, and Ludwig van Beethoven.')]"
+      ]
+     },
+     "execution_count": 87,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "question_answer_pairs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 88,
+   "id": "b8ea3f6a",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "83"
+      ]
+     },
+     "execution_count": 88,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "len(question_answer_pairs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 93,
+   "id": "1cf3c9e1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Build a prompt to provide the original query, the result and ask to summarise for the user\n",
+    "gpt_evaluator_system = '''You are WikiGPT, a helpful Wikipedia expert.\n",
+    "You will be presented with general knowledge questions our users have asked.\n",
+    "\n",
+    "Think about this step by step:\n",
+    "- You need to decide whether the answer adequately answers the question\n",
+    "- If it answers the question, you will say \"Correct\"\n",
+    "- If it doesn't answer the question, you will say one of the following:\n",
+    "    - If it couldn't answer at all, you will say \"Unable to answer\"\n",
+    "    - If the answer was provided but was incorrect, you will say \"Incorrect\" \n",
+    "- If none of these rules are met, say \"Unable to evaluate\"\n",
+    "\n",
+    "Evaluation can only be \"Correct\", \"Incorrect\", and \"Unable to evaluate\"\n",
+    "\n",
+    "Example 1:\n",
+    "\n",
+    "Question: What is the cost cap for the 2023 season of Formula 1?\n",
+    "\n",
+    "Answer: The cost cap for 2023 is 95m USD.\n",
+    "\n",
+    "Evaluation: Correct\n",
+    "\n",
+    "Example 2:\n",
+    "\n",
+    "Question: What is Thomas Dolby known for?\n",
+    "\n",
+    "Answer: Inventing electricity\n",
+    "\n",
+    "Evaluation: Incorrect\n",
+    "\n",
+    "Begin!'''\n",
+    "\n",
+    "gpt_evaluator_message = '''\n",
+    "Question: QUESTION_HERE\n",
```

Comment:
Is there any answer key or context provided here?

Reply:
I've updated at the bottom with how to respond to each of these outcomes.